mountpoint-s3/src/main.rs

```diff
@@ -78,63 +78,76 @@ fn init_tracing_subscriber(is_foreground: bool, log_directory: Option<&Path>) ->
 
 #[derive(Parser)]
 #[clap(about = "Mountpoint for Amazon S3", version = build_info::FULL_VERSION)]
-#[clap(group(ArgGroup::new("addressing-style").args(&["virtual-addressing", "path-addressing"])))]
+#[clap(group(ArgGroup::new("addressing-style").args(&["virtual_addressing", "path_addressing"])))]
 struct CliArgs {
     #[clap(help = "Name of bucket to mount")]
     pub bucket_name: String,
 
     #[clap(help = "Mount point for file system")]
     pub mount_point: PathBuf,
 
-    #[clap(short, long, help = "Log file directory. [default: $HOME/.mountpoint-s3]")]
+    #[clap(short, long, help = "Log file directory [default: $HOME/.mountpoint-s3]")]
     pub log_directory: Option<PathBuf>,
 
     #[clap(
         long,
-        help = "Prefix inside the bucket to mount. Mounts the entire bucket if unspecified."
+        help = "Prefix inside the bucket to mount [default: mount the entire bucket]",
+        help_heading = "Bucket options"
     )]
     pub prefix: Option<String>,
 
-    #[clap(long, help = "AWS region of the bucket")]
+    #[clap(
+        long,
+        help = "AWS region of the bucket [default: auto-detect region]",
+        help_heading = "Bucket options"
+    )]
     pub region: Option<String>,
 
-    #[clap(long, help = "Override S3 endpoint URL")]
+    #[clap(
+        long,
+        help = "S3 endpoint URL [default: auto-detect endpoint]",
+        help_heading = "Bucket options"
+    )]
     pub endpoint_url: Option<String>,
 
-    #[clap(long, help = "Force virtual-host-style addressing")]
+    #[clap(long, help = "Force virtual-host-style addressing", help_heading = "Bucket options")]
     pub virtual_addressing: bool,
 
-    #[clap(long, help = "Force path-style addressing")]
+    #[clap(long, help = "Force path-style addressing", help_heading = "Bucket options")]
     pub path_addressing: bool,
 
-    #[clap(long, help = "Automatically unmount on exit")]
+    #[clap(long, help = "Automatically unmount on exit", help_heading = "Mount options")]
     pub auto_unmount: bool,
 
-    #[clap(long, help = "Allow root user to access file system")]
+    #[clap(long, help = "Allow root user to access file system", help_heading = "Mount options")]
     pub allow_root: bool,
 
-    #[clap(long, help = "Allow other non-root users to access file system")]
+    #[clap(
+        long,
+        help = "Allow other non-root users to access file system",
+        help_heading = "Mount options"
+    )]
     pub allow_other: bool,
 
-    #[clap(long, help = "Desired throughput in Gbps", value_name = "N (Gbps)", value_parser = clap::value_parser!(u64).range(1..))]
+    #[clap(long, help = "Desired throughput in Gbps", value_name = "N", default_value = "10", value_parser = value_parser!(u64).range(1..), help_heading = "Client options")]
     pub throughput_target_gbps: Option<u64>,
```

Comment:
If we have default values for these, should they still be `Option<>`? Or instead just the `u64`, etc.?

Reply:
I think `Option` is a little friendlier to someone trying to use `CliArgs` programmatically.